src/System.Transactions/src/System/Transactions/CommittableTransaction.cs

```diff
@@ -42,6 +44,12 @@ internal CommittableTransaction(IsolationLevel isoLevel, TimeSpan timeout) : bas
             }
         }
 
+        // This serializtion constructor needs to be removed when we implement distributed transaction capabilities
```

Comment:
Here is how it works in Desktop:

When a Transaction (or one if its derived classes) is serialized, GetObjectData calls into the current State for the transaction. This ends up promoting the transaction, if necessary, and then calling GetObjectData on the "promoted" object, which is an OleTxTransaction object. The data that OletxTransaction returns is the "propagation token" for the DTC transaction. The "type" defined in the serializationInfo will be for the "OletxTransaction" type.

Thus, for deserialization the OletxTransaction constructor gets called and the propagation token is extracted from the serializationInfo. Then OletxTransaction.GetRealObject is called, which uses the saved propagation token to get a Transaction object corresponding to the DTC transaction. It is that Transaction object that is returned from GetRealObject.

I envision a similar scenario, where we have a public DistributedTransaction object that implementors of a distributed transaction manager would derive from and deal with serialization/deserialization.

This is why there were no serilization constructors on the various public *Transaction classes.

I can't make these "temporary" constructors protected, as suggested by serialization documentation, because the containing classes are sealed, except for Transaction.